Semantics of key attestation (#14)
thomas-fossati/draft-kat: draft-bft-rats-kat.md

# Security Considerations

## Semantics of Key Attestation

> The semantics of the KAT, i.e. what exact properties are attested, is highly implementation dependent. At the very least, the platform MUST NOT sign the KAT unless the IK's private key is generated by the Trusted Computing Base and never leaves its control.

Reviewer: > At the very least, the platform MUST NOT sign the KAT unless the IK's private key is generated by the Trusted Computing Base and never leaves its control.

I'm not sure, another option would be to make the properties of the key (e.g., whether it can be exported) explicit in the KAT.

Author: @ionut-arm I was assuming this is an ephemeral key, and therefore there is no motivation to export it and so we can assume it is protected from export. Are you aware of implementations where this is not the case?

Reviewer: I think importing/whether the key was imported was what I actually had in mind. I remember some conversations around the idea that long-lived keys that had (similarly long-lived) x509 certs associated with them might be migrated to an environment that offered attestation, and might thus benefit from both PKIX and RA.

But no, I wouldn't say I could name or identify a system where this is being done, so feel free to ignore!

Reviewer 2: I would leave the door open for an encrypted key export since some hardware security technologies can only store a limited number of keys and those key storage slots are typically reserved for persistent keys. Hence, for short-lived keys the pattern is that the encrypted key is stored somewhere on the device and can only be decrypted by the hardware security module. This use case should be fine as well. 

Here is the suggested text: 

```suggestion
The semantics of the KAT, i.e. what exact properties are attested, is highly implementation dependent. At the very least, the platform MUST NOT sign the KAT unless the IK's private key is generated by the Trusted Computing Base and is never exported in the clear.
```

Reviewer 3: please give me until the end of the week to provide some feedback here (abroad the next days)

Reviewer 4: > please give me until the end of the week to provide some feedback here (abroad the next days)

@henkbirkholz bumping this back to the top of your inbox.

> The Identity Key, when it is ephemeral, is used to secure specific protocol sessions. However in general, protocol security depends on much more than the key's security. A vulnerable protocol stack (e.g. an insecure TLS library) may leak sensitive network traffic or may not protect its integrity. As a result, the Verifier's policy must take system-level security considerations into account when deciding whether or not to accept the CAB for a particular use case. In pratice this means that in order to trust the Asserter, the Verifier must have some knowledge about its internal architecture and the security threats it is subject to.

Reviewer 4: ```suggestion
The Identity Key, when it is ephemeral, is used to secure specific protocol sessions. However in general, protocol security depends on much more than the key's security. A vulnerable protocol stack (e.g. an insecure TLS library) may leak sensitive network traffic or may not protect its integrity. As a result, the Verifier's policy must take system-level security considerations into account when deciding whether or not to accept the CAB for a particular use case. In pratice this means that in order to trust the Attester, the Verifier must have some knowledge about its internal architecture and the security threats it is subject to.
```
